plugin/pkg/auth/authorizer/rbac/bootstrappolicy/controller_policy.go

```diff
@@ -206,16 +206,31 @@ func buildControllerRoles() ([]rbacv1.ClusterRole, []rbacv1.ClusterRoleBinding)
 		},
 	})
 	addControllerRole(&controllerRoles, &controllerRoleBindings, rbacv1.ClusterRole{
-		ObjectMeta: metav1.ObjectMeta{Name: saRolePrefix + "node-controller"},
+		ObjectMeta: metav1.ObjectMeta{Name: saRolePrefix + "node-lifecycle-controller"},
 		Rules: []rbacv1.PolicyRule{
-			rbacv1helpers.NewRule("get", "list", "update", "delete", "patch").Groups(legacyGroup).Resources("nodes").RuleOrDie(),
```

Comment:
the non-cloud node controller never deletes node objects?

Reply:
It used to, but in https://github.com/kubernetes/kubernetes/pull/70344 we split the node deletion logic into a separate controller so it can be easily ported into the cloud controller manager.